controllers/designate_controller.go

```diff
@@ -1082,6 +1077,73 @@ func (r *DesignateReconciler) reconcileNormal(ctx context.Context, instance *des
 	return ctrl.Result{}, nil
 }
 
+func (r *DesignateReconciler) handleConfigMap(ctx context.Context, helper *helper.Helper, instance *designatev1beta1.Designate, configMapName string, labels map[string]string) (*corev1.ConfigMap, error) {
+	Log := r.GetLogger(ctx)
+
+	nodeConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: instance.GetNamespace(),
+			Labels:    labels,
+		},
+		Data: make(map[string]string),
+	}
+
+	// Look for existing config map and if exists, read existing data and match
+	// against nodes.
+	foundMap := &corev1.ConfigMap{}
+	err := helper.GetClient().Get(ctx, types.NamespacedName{Name: configMapName, Namespace: instance.GetNamespace()}, foundMap)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			Log.Info(fmt.Sprintf("Ip map %s doesn't exist, creating.", configMapName))
+		} else {
+			return nil, err
+		}
+	} else {
+		Log.Info("Retrieved existing map, updating..")
+		nodeConfigMap.Data = foundMap.Data
+	}
+
+	return nodeConfigMap, nil
+}
+
+func (r *DesignateReconciler) allocatePredictableIPs(ctx context.Context, predictableIPParams *designate.NADIpam, nodeNames []string, existingMap map[string]string, allocatedIPs map[string]bool) (map[string]string, map[string]bool, error) {
```

Comment:
nit: do you mind changing the arg name "nodeNames" to something else as it doesn't really match anymore?

Reply:
When we call that function for mdns, we do get the node list and pass their names. I can change it to ip_holders, maybe?
because afterwards we get something like this:
~ > oc get configmaps designate-mdns-ip-map -o json | jq -r '.data'; oc get configmaps designate-bind-ip-map -o json | jq -r '.data'
{
  "mdns_crc-vlf7c-master-0": "172.28.0.97"
}
{
  "bind_address_0": "172.28.0.98",
  "bind_address_1": "172.28.0.99"
}

Reply:
Right, it's just a nit that that in the bind case it isn't node names anymore.